### DISCUSSION TITLE
Custom Add to Cart Widget broken for Product Variations

### DISCUSSION THREAD

**Discussion by alexnasla** [elementor/elementor#21300] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I would like to use the custom add-to-cart button and pre-define which product variation is attached to that add-to-cart button. So when the user clicks on add to cart on that specific button on that page, it adds the variation I had previously pre-defined to the cart either with AJAX or by sending the user to the cart page. Currently, the Custom Add to cart widget is completely broken with product variations and does not do anything at all when pressed. Likely because it doesn't know what variation to add since there is nowhere to define it?

### Describe the solution you'd like

The solution I propose is straightforward. In the custom add to cart widget settings, allow the user to select the product they want. And if that product has variations, allow them to select which variation should be connected to that button. When the Add to cart button is pressed, it either sends the user to the cart page or if they have AJAX add to cart enabled, adds it to the cart that way. No user selection of variations, no showing of the different variation options. All that would be customized and designed with Elementor in other ways.  It just needs to be an add to cart button and nothing else to the customer. 

### Describe alternatives you've considered

I have been trying to work around this problem for a couple years now and there is no good alternative other than having no add to cart button at all on the site. I just have Buy Now buttons. Which is very inconvenient to the user as they have to keep leaving checkout to go back to the shop page to add more items.

**Comment by Bamstian** [comment]
Same here: There are several settings showing when editing the Element unter Tab "Variations" but all of them are useless since they don´t do anything. 

Please fix!

**Comment by voxtermen** [comment]
Hi! I join you, a similar problem

**Comment by hasif5** [comment]
I'm surprised this issue is still unresolved, and I haven't been able to find a workaround. All caching is disabled, and the problem seems to be with Elementor Pro's Single Add to Cart widget. Has anyone found a solution or workaround for this yet?
